pkg/volume/plugins.go

```diff
@@ -319,6 +323,15 @@ type KubeletVolumeHost interface {
 	// SetKubeletError lets plugins set an error on the Kubelet runtime status
 	// that will cause the Kubelet to post NotReady status with the error message provided
 	SetKubeletError(err error)
+
+	// GetInformerFactory returns the informer factory for CSIDriverLister
+	GetInformerFactory() informers.SharedInformerFactory
```

Comment:
Why not:
```
type KubeletVolumeHost interface {
    VolumeHost
```
And then have kubelet depend on the KubeletVolumeHost interface? Same with the attach detach volume host. All the runtime checking `..., ok := ...(KubeletVolumeHost); ok` seems like it can easily be caught at compile time.

Reply:
I think this will require a little more refactoring of the Volume interfaces before we can do this.  Right now both controller and node methods are included in the same interface, and some calls are called by both controller and kubelet.